creator-node/src/middlewares/premiumContent/premiumContentMiddleware.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  sendResponse,
+  errorResponseServerError,
+  errorResponseForbidden,
+  errorResponseBadRequest
+} from '../../apiHelpers'
+import { recoverWallet } from '../../apiSigning'
+import { NextFunction, Request, Response } from 'express'
+import { isPremiumContentMatch } from '../../premiumContent/helpers'
+import { PremiumContentType } from '../../premiumContent/types'
+import { getAllRegisteredDNodes } from '../../utils'
+import type Logger from 'bunyan'
+import { Redis } from 'ioredis'
+
+/**
+ * Middleware to validate requests to get premium content.
+ */
+export const premiumContentMiddleware = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const {
+      signedDataFromDiscoveryNode,
+      signatureFromDiscoveryNode,
+      signedDataFromUser,
+      signatureFromUser
+    } = req.headers
+
+    if (
+      !signedDataFromDiscoveryNode ||
+      !signatureFromDiscoveryNode ||
+      !signedDataFromUser ||
+      !signatureFromUser
+    ) {
+      return sendResponse(
+        req,
+        res,
+        errorResponseBadRequest('Missing request headers.')
+      )
+    }
+
+    const logger = (req as any).logger as Logger
+
+    const discoveryNodeWallet = recoverWallet(
```

Comment:
At some point here, I think we want some explicit checks as to whether the signedDataFromDiscoveryNode is the shape we expect. Probably a step that JSON.parses it, and then validates

Reply:
that happens below